pkg/operator/ceph/cluster/controller.go

```diff
@@ -391,33 +464,27 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 	} else {
 		err := c.checkIfVolumesExist(cluster)
 		if err != nil {
-			opcontroller.UpdateCondition(c.context, c.namespacedName, cephv1.ConditionDeleting, corev1.ConditionFalse, "ClusterDeleting", err.Error())
-			logger.Errorf("failed to check if volumes exist. %v", err)
-			return opcontroller.WaitForRequeueIfFinalizerBlocked, false
+			return opcontroller.WaitForRequeueIfFinalizerBlocked, errors.Wrapf(err, "failed to check if volumes exist for CephCluster in namespace %q", cluster.Namespace)
 		}
 	}
 
-	// Only valid when the cluster is not external
 	if cluster.Spec.External.Enable {
 		purgeExternalCluster(c.context.Clientset, cluster.Namespace)
-		return reconcile.Result{}, true
-	}
-
-	// If the StorageClass retain policy of an encrypted cluster with KMS is Delete we also delete the keys
-	if cluster.Spec.Storage.IsOnPVCEncrypted() && cluster.Spec.Security.KeyManagementService.IsEnabled() {
+	} else if cluster.Spec.Storage.IsOnPVCEncrypted() && cluster.Spec.Security.KeyManagementService.IsEnabled() {
+		// If the StorageClass retain policy of an encrypted cluster with KMS is Delete we also delete the keys
 		// Delete keys from KMS
 		err := c.deleteOSDEncryptionKeyFromKMS(cluster)
 		if err != nil {
-			logger.Errorf("failed to delete osd encryption keys from kms. %v", err)
-			return reconcile.Result{}, true
+			// TODO: ask Seb if this is his intended behavior; should we reconcile again instead?
```

Comment:
Actually, I prefer this behavior. It's the best effort, I'm not sure if it's worth reconciling again.